src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -556,5 +556,138 @@ public AggregateContext(T seed)
+                    dt.Rows.Add(new object[] { "System.String","nchar",16 });
+                    dt.Rows.Add(new object[] { "System.DateTime","datetime",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","smalldate",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","timestamp",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","date",6 });
+                    dt.Rows.Add(new object[] { "System.DateTime","time",6 });
+                    dt.Rows.Add(new object[] { "System.Guid","uniqueidentifier",4 });
+                    dt.Rows.Add(new object[] { "System.Guid","guid",4 });
+                    return dt;
+                case "Tables":
+                    dt.Columns.AddRange(new[] {
+                        new DataColumn("TABLE_TYPE"),
+                        new DataColumn("TABLE_CATALOG"),
+                        new DataColumn("TABLE_NAME")
+                    });
+                    var typesQuery = "SELECT type as TABLE_TYPE, 'main' as TABLE_CATALOG, name as TABLE_NAME FROM sqlite_master WHERE type='table';";
+                    if(this.State != ConnectionState.Open)
+                    {
+                        this.Open();
+                    }
+                    using (var com = new SqliteCommand(typesQuery, this))
+                    using(var reader = com.ExecuteReader())
+                    {
+                        while (reader.Read()) {
+                            dt.Rows.Add(new object[] { reader.GetString(0), reader.GetString(1), reader.GetString(2) });
+                        }
+                    }
+                    return dt;
+                case "ForeignKeys":
+                    var tableQuery = "SELECT name as TABLE_NAME FROM sqlite_master WHERE type='table';";
+                    var tables = new List<string>();
+
+                    if (this.State != ConnectionState.Open)
+                    {
+                        this.Open();
+                    }
+
+                    using (var com = new SqliteCommand(tableQuery, this))
+                    using (var reader = com.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            tables.Add(reader.GetString(0));
+                        }
+                    }
+                    dt.Columns.AddRange(new[] {
+                        new DataColumn("TABLE_NAME"),
+                        new DataColumn("FKEY_TO_CATALOG"),
+                        new DataColumn("TABLE_CATALOG"),
+                        new DataColumn("FKEY_TO_TABLE"),
+                        new DataColumn("FKEY_FROM_COLUMN"),
+                        new DataColumn("FKEY_TO_COLUMN"),
+                        new DataColumn("CONSTRAINT_NAME")
+                    });
+                    foreach(var tablename in tables){
+                        var relationQuery = "pragma foreign_key_list(" + tablename + ")";
```

Comment:
Can combine this and the previous query using this syntax:
``` sql
SELECT t.name,
       fk.table,
       fk.from,
       fk.to
FROM sqlite_master t,
     pragma_foreign_key_list(t.name) fk
WHERE t.type = 'table'
```

Reply:
Nice idea, however [DB Browser for SQLite](http://sqlitebrowser.org/) doesn't support that, so I don't know how widely that is supported.